### PR TITLE
fix: set conf info when global rule is hit without matched rule

### LINKED ISSUE #3330: bug: /apisix/prometheus/metrics cannot access
### Environment

* apisix version (cmd: `apisix version`): apache/apisix-2.2 docker
* OS: (cmd: `uname -a`) Linux gateway 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
* OpenResty / Nginx version: (cmd: `nginx -V` or `openresty -V`)

### Minimal test code / Steps to reproduce the issue

1. request GET /apisix/prometheus/metrics
2. it's not accessible

### What's the actual result? (including assertion message & call stack if applicable) 
broken

```
2021/01/19 02:10:44 [error] 43#43: *10729051 failed to run header_filter_by_lua*: /usr/local/apisix/apisix/core/lrucache.lua:159: invalid value (nil) at index 1 in table for 'concat'
stack traceback:
	[C]: in function 'concat'
	/usr/local/apisix/apisix/core/lrucache.lua:159: in function 'plugin_ctx'
	/usr/local/apisix/apisix/plugins/cors.lua:168: in function 'phase_func'
	/usr/local/apisix/apisix/init.lua:164: in function 'run_plugin'
	/usr/local/apisix/apisix/init.lua:625: in function 'common_phase'
	/usr/local/apisix/apisix/init.lua:662: in function 'http_header_filter_phase'
	header_filter_by_lua:2: in main chunk, client: 10.9.6.206, server: , request: "GET /apisix/prometheus/metrics HTTP/1.1", host: "10.9.6.225"
```
### What's the expected result?
access ok

### PR DESCRIPTION
**Pull request by spacewander**
Fix #3330

Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by tokers — APPROVED** [review]

File: t/config-center-yaml/global-rule.t [hunk 1]
```diff
@@ -112,3 +112,18 @@ global_rules:
...
 #END
 --- error_code: 403
+
+
+
+=== TEST 4: common phase without matched route
+--- apisix_yaml
+global_rules:
+    -
+        id: 1
```
Comment (juzhiyuan) [inline t/config-center-yaml/global-rule.t:122]: image

File: apisix/init.lua [hunk 2]
```diff
@@ -598,14 +598,26 @@ local function common_phase(phase_name)
     end
 
     if api_ctx.global_rules then
+        local orig_conf_type = api_ctx.conf_type
```
Comment (juzhiyuan) [inline apisix/init.lua:601]: `original? origin?`

File: t/config-center-yaml/global-rule.t (hunk 1 above)
Comment (spacewander) [reply t/config-center-yaml/global-rule.t:122]: The indentation is correct. The yaml can be parsed correctly.

File: apisix/init.lua (hunk 2 above)
Comment (spacewander) [reply apisix/init.lua:601]: `orig_` is a common abbr for `original_`, for example:
https://github.com/apache/apisix/blob/15528d815716fa907abd595f989975f12932451a/apisix/core/table.lua#L109

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**